podpac/core/test/test_node.py

```diff
@@ -215,208 +215,283 @@ def test_create_output_array_dtype(self):
+        assert not self.node.has_cache('d', coordinates=self.coords)
+
+    def test_rem_key_coordinates(self):
+        self.node.put_cache(0, 'a')
+        self.node.put_cache(0, 'b')
+        self.node.put_cache(0, 'a', coordinates=self.coords)
+        self.node.put_cache(0, 'c', coordinates=self.coords)
+        self.node.put_cache(0, 'c', coordinates=self.coords2)
+        self.node.put_cache(0, 'd', coordinates=self.coords)
+
+        self.node.rem_cache(key='a', coordinates=self.coords)
+
+        assert self.node.has_cache('a')
+        assert not self.node.has_cache('a', coordinates=self.coords)
+        assert self.node.has_cache('b')
+        assert self.node.has_cache('c', coordinates=self.coords)
+        assert self.node.has_cache('c', coordinates=self.coords2)
+        assert self.node.has_cache('d', coordinates=self.coords)
+
+class TestCachePropertyDecorator(object):
+    def test_cache_property_decorator(self):
+        class Test(podpac.Node):
+            a = tl.Int(1).tag(attr=True)
+            b = tl.Int(1).tag(attr=True)
+            c = tl.Int(1)
+            d = tl.Int(1)
+
+            @podpac.core.node.cached_property('a2', 'a')
+            def a2(self):
+                """a2 docstring"""
+                return self.a * 2
+
+            @podpac.core.node.cached_property('b2')
+            def b2(self):
+                """ b2 docstring """
+                return self.b * 2
+
+            @podpac.core.node.cached_property('c2', 'c')
+            def c2(self):
+                """ c2 docstring """
+                return self.c * 2
+
+            @podpac.core.node.cached_property('d2')
+            def d2(self):
+                """ d2 docstring """
+                return self.d * 2
+            
+        t = Test(cache_type='disk')
+        t2 = Test(cache_type='disk')
+        t.rem_cache()
+        t2.rem_cache()
+
+        try: 
+            t.get_cache('a2')
+            raise Exception("Cache should be cleared.")
+        except podpac.NodeException:
+            pass
 
-    def test_cache_dir(self):
-        n = Node()
-        assert isinstance(n.cache_dir, six.string_types)
-        assert n.cache_dir.endswith('Node')
-        assert 'cache' in n.cache_dir
+        assert t.a2() == 2
+        assert t.b2() == 2
+        assert t.c2() == 2
+        assert t.d2() == 2
+        assert t2.a2() == 2
+        assert t2.b2() == 2
+        assert t2.c2() == 2
+        assert t2.d2() == 2
+
+        t.a = 2
+        assert t.a2() == 4
+        t.b = 2
+        assert t.b2() == 4  # This happens because the node definition changed
+        t.rem_cache()
+        assert t.c2() == 2  # This forces the cache to update based on the new node definition
+        assert t.d2() == 2  # This forces the cache to update based on the new node definition
+        t.c = 2
+        assert t.c2() == 4  # This happens because of depends
+        t.d = 2
+        assert t.d2() == 2  # No depends, and doesn't have a tag
```

Comment:
This test actually really highlights a user error... but this is the specified behaviour. Somewhat similar to the @tl.default behaviour -- only called once, not updated.